plugins/python/filters/PythonFilter.cpp

```diff
@@ -93,8 +94,15 @@ void PythonFilter::addArgs(ProgramArgs& args)
 
 void PythonFilter::addDimensions(PointLayoutPtr layout)
 {
-    for (const std::string& s : m_args->m_addDimensions)
-        layout->registerOrAssignDim(s, pdal::Dimension::Type::Double);
+    for (const std::string& s : m_args->m_addDimensions) {
```

Comment:
The code should provide feedback to the user if the syntax of the argument is bad.  Here, spec size should only be 1 or 0 and a check should be made to make sure spec[1] is a valid type.